packages/components/components/elvis-context-menu/src/react/elvia-context-menu.tsx

```diff
@@ -17,98 +30,115 @@ export const ContextMenu: React.FC<ContextMenuProps> = ({
   trigger,
   inlineStyle,
   display = 'inline-block',
+  anchorPosition,
   className,
   webcomponent,
   ...rest
 }) => {
-  const popoverRef = useRef<HTMLDivElement>(null);
   const triggerRef = useRef<HTMLDivElement>(null);
-  const { trapFocus, releaseFocusTrap } = useFocusTrap();
-  const [prevFocusedElement, setPrevFocusedElement] = useState<HTMLElement>();
-  const [fadeOut, setFadeOut] = useState(false);
+  const [isOpen, setIsOpen] = useState(isShowing);
 
   useSlot('trigger', webcomponent, { ref: triggerRef });
 
-  const { isShowing: isOverlayShowing = false, setIsShowing: setIsOverlayShowing } = useConnectedOverlay(
-    triggerRef,
-    popoverRef,
-    {
-      horizontalPosition: mapPositionToHorizontalPosition(horizontalPosition),
-      verticalPosition: verticalPosition,
-      alignWidths: false,
+  const updateOpenState = (newState: boolean) => {
+    setIsOpen(newState);
+    if (newState) {
+      handleOnOpen();
+    } else {
+      handleOnClose();
+    }
+  };
+
+  const { refs, floatingStyles, context } = useFloating({
+    placement: mapPosition(verticalPosition, horizontalPosition),
+    open: isOpen,
+    onOpenChange: updateOpenState,
+    elements: {
+      reference: triggerRef.current,
     },
-  );
+    middleware: [offset({ mainAxis: 8 }), flip(), shift()],
+    whileElementsMounted: autoUpdate,
+    transform: false,
+  });
+
+  // This hook enables the floating element to be positioned based on the anchorPosition prop
+  const clientPoint = useClientPoint(context, {
+    enabled: !!anchorPosition,
```

Comment:
Is this "safe enough", given that an empty object still returns true?

Reply:
I can add checks that `anchorPosition.top` and `anchorPosition.left` are set if we want too, but if someone sends in an empty object I would see it as a user error 🤔

Reply:
Good point, lets keep it simple like this